modules/publicize/publicize.php

```diff
@@ -510,6 +510,215 @@ function test_publicize_conns() {
+				if ( ! apply_filters( 'wpas_submit_post?', true, $post_id, $service_name, $connection_data ) ) {
+					continue;
+				}
+
+				// Should we be skipping this one?
+				$skip = (
+					(
+						! empty( $post )
+						&&
+						in_array( $post->post_status, array( 'publish', 'draft', 'future' ) )
+						&&
+						(
+							// New flags
+							get_post_meta( $post_id, $this->POST_SKIP . $unique_id, true )
+							||
+							// Old flags
+							get_post_meta( $post->ID, $this->POST_SKIP . $service_name )
+						)
+					)
+					||
+					(
+						is_array( $connection )
+						&&
+						isset( $connection_meta['external_id'] ) && ! empty( $service_id_done[ $service_name ][ $connection_meta['external_id'] ] )
+					)
+				);
+
+				// If this one has already been publicized to, don't let it happen again.
+				$toggleable = ! $done && ! $all_done;
+
+				// Determine the state of the checkbox (on/off) and allow filtering.
+				$enabled = $done || ! $skip;
+				/**
+				 * Filter the checkbox state of each Publicize connection appearing in the post editor.
+				 *
+				 * @module publicize
+				 *
+				 * @since 2.0.1
+				 *
+				 * @param bool $enabled Should the Publicize checkbox be enabled for a given service.
+				 * @param int $post_id Post ID.
+				 * @param string $service_name Service name.
+				 * @param array $connection Array of connection details.
+				 */
+				$enabled = apply_filters( 'publicize_checkbox_default', $enabled, $post_id, $service_name, $connection );
+
+				/**
+				 * If this is a global connection and this user doesn't have enough permissions to modify
+				 * those connections, don't let them change it.
+				 */
+				if ( ! $done && ( 0 == $connection_data['user_id'] && ! current_user_can( $this->GLOBAL_CAP ) ) ) {
+					$toggleable = false;
+
+					/**
+					 * Filters the checkboxes for global connections with non-prilvedged users.
+					 *
+					 * @module publicize
+					 *
+					 * @since 3.7.0
+					 *
+					 * @param bool   $enabled Indicates if this connection should be enabled. Default true.
+					 * @param int    $post_id ID of the current post
+					 * @param string $service_name Name of the connection (Facebook, Twitter, etc)
+					 * @param array  $connection Array of data about the connection.
+					 */
+					$enabled = apply_filters( 'publicize_checkbox_global_default', $enabled, $post_id, $service_name, $connection );
+				}
+
+				// Force the checkbox to be checked if the post was DONE, regardless of what the filter does.
+				if ( $done ) {
+					$enabled = true;
+				}
+
+				$label  = sprintf(
+					/* translators: %1$s: Service Name (Facebook, Twitter, ...), %2$s: Username on Service (@jetpack, ...) */
+					__( '%1$s: %2$s', 'jetpack' ),
+					$this->get_service_label( $service_name ),
+					$this->get_display_name( $service_name, $connection )
+				);
+
+				$connection_list[] = array(
+					'unique_id'    => $unique_id,
+					'service_name' => $service_name,
+					'label'        => $label,
```

Comment:
#9955 used to also provide [`displayName`](https://github.com/Automattic/jetpack/pull/9955/files#diff-2619864631f3665a877cdd9148da0072R594) (i.e. `label` without service label). This came in handy for Calypso, where we indicate the service name through icons:

![bildschirmfoto von 2018-10-25 21-05-06](https://user-images.githubusercontent.com/96308/47524263-68d2aa80-d89a-11e8-88dd-cec4cff88314.png)

Can we add back `display_name`, or even ditch `label` (and move concatenation with the service label to `ui.php`)?

Reply:
Here's a [workaround](https://github.com/Automattic/wp-calypso/pull/28102/commits/073f9561df6122855676fd5a60286fe3e8c29dca), but I'd rather not :)

Reply:
Yes! I'll get this sorted shortly.

Reply:
Done in 3995e0850

Reply:
(And tests fixed in c4f653cbe :))

Reply:
Awesome, thanks!